### PR TITLE
CRLNumber needs to be a class for reasons

### PR DESCRIPTION
**Pull request by reaperhulk**

### CODE REVIEW

File: docs/x509/reference.rst [hunk 1]
```diff
@@ -1695,6 +1695,27 @@ X.509 Extensions
 
         :type: int
 
+.. class:: CRLNumber(crl_number)
+
+    .. versionadded:: 1.2
+
+    The CRL number is a non-critical CRL extension that conveys a
```
Comment (alex) [inline docs/x509/reference.rst]: Err, what does non-critical mean in this context?

Comment (reaperhulk) [inline docs/x509/reference.rst]: That comes from the RFC, but it means it should not be marked as critical. We could just strip it though.

Comment (alex) [inline docs/x509/reference.rst]: I'd drop it, it kind of makes no sense.

File: docs/x509/reference.rst [hunk 2]
```diff
@@ -1695,6 +1695,27 @@ X.509 Extensions
...
+    monotonically increasing sequence number for a given CRL scope and
+    CRL issuer. This extension allows users to easily determine when a
+    particular CRL supersedes another CRL. :rfc:`5280` requires that this
+    extension be present in conforming CRLs.
```
Comment (alex) [inline docs/x509/reference.rst]: So, it has to be included but should never be marked critical? Ugh, the IETF is teh worst.

Comment (reaperhulk) [inline docs/x509/reference.rst]: yuuup

**Comment by alex** [comment]
besides those language tweeks this looks ok pending tests

File: src/cryptography/x509/extensions.py [hunk 3]
```diff
@@ -109,6 +109,31 @@ def __repr__(self):
 
 
 @utils.register_interface(ExtensionType)
+class CRLNumber(object):
+    oid = ExtensionOID.CRL_NUMBER
+
+    def __init__(self, crl_number):
+        if not isinstance(crl_number, six.integer_types):
+            raise TypeError("crl_number must be an integer")
```
Comment (alex) [inline src/cryptography/x509/extensions.py]: This line is untested

**Comment by reaperhulk** [comment]
Jenkins, retest this please

**Pull request merged**